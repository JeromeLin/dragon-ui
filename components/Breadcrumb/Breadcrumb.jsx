import React, { Component, cloneElement } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class Breadcrumb extends Component {
  render() {
    const {
      className, separator, children, style
    } = this.props;

    const cls = classnames({
      'ui-breadcrumb': true,
      [className]: !!className
    });

    // eslint-disable-next-line
    const items = React.Children.map(children, (element, index) => {
      return cloneElement(element, {
        separator,
        key: index
      });
    });

    return (
      <div className={cls} style={style}>
        {items}
      </div>
    );
  }
}

Breadcrumb.propTypes = {
  separator: PropTypes.string,
};

Breadcrumb.defaultProps = {
  separator: '/'
};

export default Breadcrumb;
