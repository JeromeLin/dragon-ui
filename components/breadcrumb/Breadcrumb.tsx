import React, { Component, cloneElement, ReactElement, HTMLAttributes } from 'react';
import classnames from 'classnames';

export type BreadcrumbProps = {
  prefixCls?: string;
  separator?: string;
} & HTMLAttributes<HTMLDivElement>;

class Breadcrumb extends Component<BreadcrumbProps, {}> {
  static displayName = 'Breadcrumb';

  static defaultProps = {
    prefixCls: 'zw-breadcrumb',
    separator: '/',
  };

  static Item;

  render() {
    const { prefixCls, className, separator, children, ...restProps } = this.props;
    const cls = classnames(prefixCls, className);

    const items = React.Children.map(children, (element, index) => {
      return cloneElement(element as ReactElement<any>, {
        separator,
        key: index,
      });
    });

    return (
      <div className={cls} {...restProps}>
        {items}
      </div>
    );
  }
}

export default Breadcrumb;
