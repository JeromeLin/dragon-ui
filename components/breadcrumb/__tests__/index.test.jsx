import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Breadcrumb from '../index';

describe('Breadcrumb', () => {
  it('renders basic Breadcrumb correctly', () => {
    const wrapper = render(
      <Breadcrumb>
        <Breadcrumb.Item>一级菜单</Breadcrumb.Item>
        <Breadcrumb.Item>二级菜单</Breadcrumb.Item>
        <Breadcrumb.Item href="https://t.zhongan.com">三级菜单</Breadcrumb.Item>
      </Breadcrumb>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Breadcrumb with customized separator correctly', () => {
    const wrapper = render(
      <Breadcrumb separator=">">
        <Breadcrumb.Item>一级菜单</Breadcrumb.Item>
        <Breadcrumb.Item>二级菜单</Breadcrumb.Item>
        <Breadcrumb.Item href="https://t.zhongan.com">三级菜单</Breadcrumb.Item>
      </Breadcrumb>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
