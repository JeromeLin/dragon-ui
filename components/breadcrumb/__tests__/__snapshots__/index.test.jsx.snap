// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders Breadcrumb with customized separator correctly 1`] = `
<div
  class="zw-breadcrumb"
>
  <span
    class="zw-breadcrumb-item"
  >
    <span
      class="zw-breadcrumb-item__content"
    >
      一级菜单
    </span>
    <span
      class="zw-breadcrumb-item__separator"
    >
      &gt;
    </span>
  </span>
  <span
    class="zw-breadcrumb-item"
  >
    <span
      class="zw-breadcrumb-item__content"
    >
      二级菜单
    </span>
    <span
      class="zw-breadcrumb-item__separator"
    >
      &gt;
    </span>
  </span>
  <span
    class="zw-breadcrumb-item"
  >
    <a
      class="zw-breadcrumb-item__link"
      href="https://t.zhongan.com"
    >
      三级菜单
    </a>
    <span
      class="zw-breadcrumb-item__separator"
    >
      &gt;
    </span>
  </span>
</div>
`;

exports[`Breadcrumb renders basic Breadcrumb correctly 1`] = `
<div
  class="zw-breadcrumb"
>
  <span
    class="zw-breadcrumb-item"
  >
    <span
      class="zw-breadcrumb-item__content"
    >
      一级菜单
    </span>
    <span
      class="zw-breadcrumb-item__separator"
    >
      /
    </span>
  </span>
  <span
    class="zw-breadcrumb-item"
  >
    <span
      class="zw-breadcrumb-item__content"
    >
      二级菜单
    </span>
    <span
      class="zw-breadcrumb-item__separator"
    >
      /
    </span>
  </span>
  <span
    class="zw-breadcrumb-item"
  >
    <a
      class="zw-breadcrumb-item__link"
      href="https://t.zhongan.com"
    >
      三级菜单
    </a>
    <span
      class="zw-breadcrumb-item__separator"
    >
      /
    </span>
  </span>
</div>
`;
